include/ppx/application.h

```diff
@@ -323,6 +323,33 @@ struct ApplicationSettings
             uint32_t     imageCount  = 2;
         } swapchain;
     } grfx;
+
+    // Default values for standard knobs
+    struct StandardKnobsDefaultValue
+    {
+        std::vector<std::string> assetsPaths     = {};
+        std::vector<std::string> configJsonPaths = {};
```

Comment:
Thinking about it, would these two flags `--asset-path` and `--config-json-path` ever require default values? They seem to be flags that would only be used in non-standard usage of benchmarks/projects.

Reply:
It has benefit to specify all of them at the same place. If there's use case for them to be updated, they will remain null value

Reply:
I see the point about it being nicer to have defaults in the same place, I'm just thinking that these 2 flags should never have non-empty defaults. Any knob value in the config file should just become that knob's default value (avoid having a default config file) and then assets used by default ought to be in the assets folder. I think these two flags should be excluded from `ApplicationSettings` to prevent application writers from setting defaults for them if this is an antipattern that we'd like to avoid.

Reply:
If you want to prevent application writer from setting it, put a `const` modifier on it.
Also there does exist usecase for default asset paths/config paths, although to make it useful you need to make the last one take higher priority.
i.e. `--asset-path=platform-default --asset-path=application-default --asset-path=user-override1 --asset-path=user-override2`

Reply:
I'd prefer to have default in flags and get rid of `Application::InitializeAssetDirs`

Reply:
i will address the asset dir in a separate PR